File: 1046-last-stone-weight/1046-last-stone-weight.cpp
class Solution {
public:
    int lastStoneWeight(vector<int>& stones) {
        priority_queue<int> ston(stones.begin(),stones.end());
        
        while(ston.size()>1){
            int a = ston.top();
            ston.pop();
            int b = ston.top();
            ston.pop();
            if(a!=b){
                ston.push(abs(a-b));
            }
        }
        return ston.empty()? 0 : ston.top();
    }
};

File: 1046-last-stone-weight/1046-last-stone-weight.py
class Solution:
    def lastStoneWeight(self, stones: List[int]) -> int:
        stones = [-1*x for x in stones]
        heapify(stones)
        
        while len(stones) > 1:
            a = heappop(stones)*-1
            b = heappop(stones)*-1
        
            if a != b:
                c = max(a,b) - min(a,b)
                heappush(stones,c*-1)
                
        return stones[-1]*-1 if stones else 0
